# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1286638942 14400
#      Sat Oct 09 11:42:22 2010 -0400
# Node ID d748acc75658e9ae9e79540da33244398b2e5ed6
# Parent  8b9aeb20c03c55aeb873a90b79af33b1d7342cf0
gl-render.cc: Treat images with scalar unique(x/ydata) like Matlab.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-10-09  Ben Abbott <bpabbott@mac.com>
+
+	* gl-render.cc: Treat images with scalar unique(x/ydata) like Matlab.
+
 2010-10-08  John W. Eaton  <jwe@octave.org>
 
 	* octave.cc (maximum_braindamage):
 	Set do_braindead_shortcircuit_evaluation to true.
 	Disable Octave:possible-matlab-short-circuit-operator warning.
 	* oct-parse.yy (if_cmd_list1, elseif_clause, loop_command):
 	Mark conditions in IF and WHILE statements for braindead
 	short-circuit behavior.
diff --git a/src/gl-render.cc b/src/gl-render.cc
--- a/src/gl-render.cc
+++ b/src/gl-render.cc
@@ -2698,18 +2698,25 @@ opengl_renderer::draw_text (const text::
 void
 opengl_renderer::draw_image (const image::properties& props)
 {
   octave_value cdata = props.get_color_data ();
   dim_vector dv (cdata.dims ());
   int h = dv(0), w = dv(1);
   bool ok = true;
   
-  const Matrix x = props.get_xdata ().matrix_value ();
-  const Matrix y = props.get_ydata ().matrix_value ();
+  Matrix x = props.get_xdata ().matrix_value ();
+  Matrix y = props.get_ydata ().matrix_value ();
+
+  if (w > 1 && x(1) == x(0)) 
+    x(1) = x(1) + (w-1);
+
+  if (h > 1 && y(1) == y(0)) 
+    y(1) = y(1) + (h-1);
+
   const ColumnVector p0 = xform.transform (x(0), y(0), 0);
   const ColumnVector p1 = xform.transform (x(1), y(1), 0);
 
   // image pixel size in screen pixel units
   float pix_dx, pix_dy;
   // image pixel size in normalized units
   float nor_dx, nor_dy;
 
